### PR TITLE
bug-fix/internal-issue-#724

### PR DESCRIPTION
**Pull request by gnusi**
disallow using V8 dependent function in SEARCH statement
covered by newly added JS tests

### CODE REVIEW

**Comment by gnusi** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/10768/

File: tests/js/common/aql/aql-view-arangosearch-cluster.inc [hunk 1]
```diff
@@ -187,6 +187,14 @@
         }
       },
 
+      testV8FunctionInSearch: function () {
+        try {
+          db._query("FOR doc IN CompoundView SEARCH doc.a == 'foo' && APPLY('LOWER', 'abc') == 'ABC' OPTIONS { waitForSync: true } RETURN doc");
+        } catch (e) {
```
Comment (Dronplane) [inline tests/js/common/aql/aql-view-arangosearch-cluster.inc:193]:
```suggestion
          fail();
        } catch (e) {
```

File: tests/js/common/aql/aql-view-arangosearch-noncluster.js [hunk 2]
```diff
@@ -150,6 +150,14 @@ function iResearchAqlTestSuite () {
       }
     },
 
+    testV8FunctionInSearch: function () {
+      try {
+        db._query("FOR doc IN CompoundView SEARCH doc.a == 'foo' && APPLY('LOWER', 'abc') == 'ABC' OPTIONS { waitForSync: true } RETURN doc");
+      } catch (e) {
```
Comment (Dronplane) [inline tests/js/common/aql/aql-view-arangosearch-noncluster.js:156]:
```suggestion
        fail();
      } catch (e) {
```

**Comment by gnusi** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/10769/

**Comment by gnusi** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/10770/

**Comment by gnusi** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/10774/

**Pull request merged**